src/dotnet/commands/dotnet-projectmodel-server/ProjectManager.cs

```diff
@@ -200,8 +200,10 @@ private bool ProcessMessage()
                     _configure.Value = message.Payload.GetValue("Configuration");
                     break;
                 case MessageTypes.RefreshDependencies:
+                    _refreshDependencies.Value = false;
+                    break;
                 case MessageTypes.RestoreComplete:
-                    _refreshDependencies.Value = 0;
+                    _refreshDependencies.Value = message.Payload.HasValues ? message.Payload.Value<bool>("Reset") : true;
```

Comment:
Will that mean that its up to tooling to decide when cache should be cleaned? Meaning tooling would have to monitor packages cache folder and then send PMS a Reset flag?